### PR TITLE
Do not return error message for hover and symbols req

### PR DESCRIPTION
**Pull request by alanz**
IF the module is not currently loaded, simply return an empty result
for hover request and document symbols request.

Affects #416, #460, #462

I am not sure that it closes any of them though.

### CODE REVIEW

File: src/Haskell/Ide/Engine/Dispatcher.hs [hunk 1]
```diff
@@ -23,7 +23,7 @@ data DispatcherEnv = DispatcherEnv
   , docVersionTVar     :: !(TVar (Map.Map Uri Int))
   }
 
-dispatcherP :: forall void. DispatcherEnv -> TChan PluginRequest -> IdeGhcM void
+dispatcherP :: DispatcherEnv -> TChan PluginRequest -> IdeGhcM ()
```
Comment (wz1000) [inline src/Haskell/Ide/Engine/Dispatcher.hs]: Why is this needed? The old type is more descriptive, as it states the the function never "returns", i.e. loops forever.

The old type is also strictly more general than the new type(IdeGhcM () is a subtype of forall void. IdeGhcM void)

**Comment by wz1000** [comment]
This seems good for now, but we really need a ModuleStatus flag, that tracks whether the module compiled successfully or not, whether the typechecked module we have is fresh or not etc.

**Comment by alanz** [comment]
I agree on the module status, it ties into the life cycle being explicit as for #460 .

But this sorts out an initial annoyance, paving the way for that.

I also want to move all that Hover processing into the `HieExtras` module.
